webapp/src/components/FileMultiSelectDropdown.vue

```diff
@@ -0,0 +1,350 @@
+          >
+            {{ index + 1 }}. {{ getFileName(fileId) }}
+            <span v-if="isLive(fileId)" class="live-indicator ml-2">
+              <b>Live</b> ({{ lastModified(fileId) }})
+            </span>
+          </li>
+          <li v-if="!modelValue.length" class="list-group-item disabled">No files selected.</li>
+        </ul>
+      </div>
+
+      <div class="order-buttons">
+        <button
+          class="btn btn-light mb-2"
+          :disabled="!canMoveUp"
+          aria-label="Move selected file up"
+          @click="moveUp()"
+        >
+          &#9650;
+        </button>
+        <button
+          class="btn btn-light"
+          :disabled="!canMoveDown"
+          aria-label="Move selected file down"
+          @click="moveDown()"
+        >
+          &#9660;
+        </button>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { updateBlockFromServer } from "@/server_fetch_utils.js"; // Assuming this path is correct
+
+export default {
+  name: "MultiFileSelector",
+  props: {
+    modelValue: {
+      // Now expects an array of file IDs
+      type: Array,
+      default: () => [],
+    },
+    item_id: { type: String, required: true },
+    block_id: { type: String, required: true },
+    extensions: {
+      type: Array, // array of strings, file extensions
+      default: () => [""], // show all files (if empty string means all)
+    },
+    updateBlockOnChange: {
+      type: Boolean,
+      default: false,
+    },
+    mainLabel: {
+      type: String,
+      default: "Select and order files:",
+    },
+  },
+  emits: ["update:modelValue"],
+  data() {
+    return {
+      selectedAvailable: null, // ID of the highlighted file in the available list
+      selectedSelected: null, // ID of the highlighted file in the selected list
+      selectedSelectedIndex: -1, // Index of the highlighted file in the selected list
+    };
+  },
+  computed: {
+    all_files_map() {
+      // Create a map for potentially faster lookups if the list is large
+      const fileMap = new Map();
+      const files = this.$store.state.all_item_data[this.item_id]?.files || [];
+      files.forEach((file) => fileMap.set(file.immutable_id, file));
+      return fileMap;
+    },
+    all_available_file_ids() {
+      // All file IDs potentially available for this item
+      const allIds = this.$store.state.all_item_data[this.item_id]?.file_ObjectIds || [];
+      const showAll =
+        this.extensions.length === 0 || (this.extensions.length === 1 && this.extensions[0] === "");
+
+      return allIds.filter((file_id) => {
+        const file = this.all_files_map.get(file_id);
+        if (!file) return false; // Skip if file details not found
+
+        if (showAll) return true; // Show all if extensions are empty or just ""
+
+        const filename = file.name?.toLowerCase() || "";
+        return this.extensions.some((extension) => filename.endsWith(extension.toLowerCase()));
```

Comment:
I guess I was more worried about file selection on the UI side e.g., this bit, but you already have a custom component for it that works for UV Vis this way. I'll try to get around to reviewing the UI code tomorrow!